assets/blocks/core-image/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Spinner } from '@wordpress/components';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import './style.scss';
+
+/**
+ * Add image credit meta to core/image block attributes
+ */
+addFilter(
+	'blocks.registerBlockType',
+	'newspack-plugin/register-hook/core-image',
+	( settings, name ) => {
+		if ( name !== 'core/image' ) {
+			return settings;
+		}
+		return {
+			...settings,
+			attributes: {
+				...settings.attributes,
+				meta: {
+					type: 'object',
+					default: {},
+				},
+			},
+		};
+	}
+);
+
+/**
+ * Display spinner and lock post saving until meta attributes are added to block
+ */
+const AttributesLoader = props => {
+	const imageId = props.attributes.id;
+	const { lockPostSaving, unlockPostSaving } = useDispatch( 'core/editor' );
+	const { meta } = useSelect( select => select( 'core' ).getMedia( imageId ), [ imageId ] ) ?? {
+		meta: {},
+	};
+	// Meta added, proceed
+	if ( Object.keys( meta ).length ) {
+		props.setAttributes( { meta } );
+		unlockPostSaving( 'attachment-meta-empty' );
+	} else {
+		lockPostSaving( 'attachment-meta-empty' );
+	}
```

Comment:
These calls to `unlockPostSaving` and `lockPostSaving` happen on each render, which is not optimal. That code block should happen in a `useEffect` hook, only called if the value of `meta` has changed.

Reply:
Nice catch, fixed.